lib/OpenQA/Task/Asset/Download.pm

```diff
@@ -100,8 +100,9 @@ sub _download {
             $ctx->debug(qq{Download of "$assetpath" successful});
         }
     };
-    unless ($downloader->download($url, $assetpath, $options)) {
-        $ctx->error(my $msg = qq{Downloading "$url" failed because of too many download errors});
+    my $err = $downloader->download($url, $assetpath, $options);
+    if ($err) {
```

Comment:
You can combine the two above lines, e.g. `if (my $err = ...`

Reply:
I've added this simplification.